libs/@blockprotocol/type-system/typescript/src/native/url.ts

```diff
@@ -106,34 +107,94 @@ export const validateVersionedUrl = (
       };
     }
 
-    const index = version.search(/[^0-9]/);
-    if (index === 0) {
-      return {
-        type: "Err",
-        inner: {
-          reason: "InvalidVersion",
-          inner: [version, "invalid digit found in string"],
-        },
-      };
-    } else if (index > 0) {
-      return {
-        type: "Err",
-        inner: {
-          reason: "AdditionalEndContent",
-          inner: version.substring(index),
-        },
-      };
-    }
-
-    const versionNumber = Number(version);
-    if (versionNumber > 4294967295) {
-      return {
-        type: "Err",
-        inner: {
-          reason: "InvalidVersion",
-          inner: [version, "number too large to fit in target type"],
-        },
-      };
+    // Try to match draft version format first: "2-draft.lane.5"
```

Comment:
fwiw there's also: https://www.npmjs.com/package/semver that could be used here, additionally `[^.]` is too lax as per semver spec.

Reply:
What we could do is to parse it as `major-prerelease`, modify `major` to `${major}.0.0` and pass it on to semver, but that is an overhead.